File: routes/flow/checkout.js
var cates = require("./models/cates.js");
var res = require('./lib/resp.js');
var session = require('./lib/session.js');


var user = session.getUser();
if (!user) {
  return res.redirect('/routes/user/login.js');
}

if (request.getMethod() === 'GET') {
  var cart = session.getCart();
  if (cart.items().length == 0) {
    return res.send('购物车为空');
  }
  var allData = cates.find();
  var data = {
    tree: cates.getTree(allData),
    user: user,
    cart: cart,
  };
  res.render('./views/flow/checkout.ejs', data);
}

File: routes/flow/checkout.go
package flow

import (
	"github.com/zengming00/go-testShop/framework"
	"github.com/zengming00/go-testShop/lib"
)

func Checkout(ctx *framework.HandlerContext) {
	var user = ctx.GetUser()
	if user == nil {
		ctx.Redirect("/routes/user/login.go")
		return
	}
	if ctx.R.Method == "GET" {
		var cart = ctx.GetCart()
		if len(cart.Items()) == 0 {
			ctx.Send("购物车为空")
			return
		}
		var allData = ctx.Cates.Find()
		var data = map[string]interface{}{
			"tree": ctx.Cates.GetTree(allData),
			"user": user,
			"cart": cart,
		}
		ctx.Render("./views/flow/checkout.html", data, lib.FuncMap)
		return
	}
}
